### DISCUSSION TITLE
Hadoop SDK need more memory？

### DISCUSSION THREAD

**Discussion by neuyilan** [juicedata/juicefs#5700] [General]
Hi, I saw the following message on the official website and I want to know why?
Does this mean that when I use Hadoop SDK to read/write data, I need more memory compared to using mount fuse to read/write local files?

> JuiceFS Hadoop Java SDK may require an additional 4 * [juicefs.memory-size](https://juicefs.com/docs/cloud/hadoop/configuration#cache) off-heap memory to speed up read and write performance

**Comment by tangyoupeng** [comment]
Yes, but you can get better performance when used in Hadoop.
